services/com.mde-network.ep.toolfunctions.xtext/src/main/java/com/mdenetnetwork/ep/toolfunctions/xtext/XtextTool.java

```diff
@@ -0,0 +1,227 @@
+		// Register resource factories
+		Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().put("ecore", new EcoreResourceFactoryImpl());
+	}
+
+
+	public void run(String languageName, String baseName, String extension, String grammar, List<ProjectFile> projectFiles, OutputStream outputStream, JsonObject response) throws Exception {
+
+
+		
+		// Create Xtext project with web editor enabled
+		File projectFolder = new File(PROJECT_PATH);
+		
+		String[] args = new String[] {
+				"-targetDir=" + projectFolder.getAbsolutePath(),
+				"-languageName="+languageName,
+				"-baseName="+baseName,
+				"-extension="+extension,
+				"-enabledProjects=UIProject,GenericIDE,RuntimeTest,UITest,Web",
+				"-buildSystem=MAVEN",
+				"-sourceLayout=PLAIN",
+				"-languageServer=FATJAR",
+				"-xtextVersion=2.30.0",
+				"-projectLayout=FLAT",
+				"-javaVersion=JAVA11"
+			};
+
+		
+		CliProjectsCreatorMain.main(args);
+		
+		
+		
+		
+		// Add activity files
+		if (grammar != null) {
+			final String grammarSrcPath = languageName.replace('.', '/');
+			final String xtextGrammarPath= PROJECT_PATH + baseName + "/src/" + grammarSrcPath  + ".xtext";
+			recreateGrammarFile(xtextGrammarPath, grammar);
+		}
+		
+		//TODO add activity files
+		
+		
+		// Compress files for transfer to build server
+		
+		ZipOutputStream projectFileStream = new ZipOutputStream( new FileOutputStream(PROJECT_FILE_PATH) );
+		
+		compressDirectory(projectFolder, "", projectFileStream);
+		 
+		projectFileStream.flush();
+		projectFileStream.close();
+		 
+		
+		// Send to the build server
+		
+		final String EDITOR_SERVER_URL= "http://127.0.0.1:10001/xtext/upload";  // TODO set URL via environment variable 
+		 
+		 try {	 
+			 postProjectToUrl(new File(PROJECT_FILE_PATH), EDITOR_SERVER_URL, response);
+			 
+		 } catch (HttpHostConnectException e)  {
+		 
+			 System.err.println("Failed to connect to editor server: " + EDITOR_SERVER_URL);
+			 
+		}finally {
+			 // Cleanup created files
+			 deleteDir(projectFolder);
+			 new File(PROJECT_FILE_PATH).delete();
+		 }
+
+	}
+
+	
+	public void convertXToY(String X,  OutputStream outputStream, JsonObject response) throws Exception {
+		
+		final String emfaticFilename = "emfatic.emfatic";
+
+		String generatedY = "";
+		
+		/*-------------------------------------
+		 *  Load X 
+		 *-------------------------------------*/	
+		
+	
+
+		/*-------------------------------------
+		 *  Generate Y
+		 *-------------------------------------*/		
+
+		
+		 response.addProperty("output",  generatedY );
+	}
+	
+	
+	private void recreateGrammarFile (String path, String contents) {
+		
+	    System.out.println("Re-creating grammar file: " + path);
+	
+		try {
+			Files.writeString(Path.of(path), contents, StandardOpenOption.TRUNCATE_EXISTING);
+			
+		} catch (IOException e) {
+			
+			System.err.println("Error "+ e.toString() +" saving gramar file: " + path);
+		}
+	}
+
+	private void recreateProjectFiles () {
+		//TODO recreate project files
```

Comment:
What is the purpose of this empty function?

Reply:
Agreed I'll remove it.

Reply:
Cleaned up in 5238085fe028be75a6daea1b139452b797742ee9